clients/ui/frontend/src/app/pages/modelCatalog/utils/modelCatalogUtils.ts

```diff
@@ -118,32 +121,67 @@ export const shouldShowValidatedInsights = (
+// Define array-based filter keys (excluding USE_CASE which is single-selection)
+type ArrayFilterKey =
+  | ModelCatalogStringFilterKey.TASK
+  | ModelCatalogStringFilterKey.PROVIDER
+  | ModelCatalogStringFilterKey.LICENSE
+  | ModelCatalogStringFilterKey.LANGUAGE
+  | ModelCatalogStringFilterKey.HARDWARE_TYPE;
+
+// Type mapping for array filter values
+type ArrayFilterValueType = {
+  [ModelCatalogStringFilterKey.TASK]: ModelCatalogTask;
+  [ModelCatalogStringFilterKey.PROVIDER]: ModelCatalogProvider;
+  [ModelCatalogStringFilterKey.LICENSE]: ModelCatalogLicense;
+  [ModelCatalogStringFilterKey.LANGUAGE]: AllLanguageCode;
+  [ModelCatalogStringFilterKey.HARDWARE_TYPE]: string;
+};
+
+// Type guard to check if a value is an array of the expected type
+const isArrayOfValues = <T>(value: unknown): value is T[] => Array.isArray(value);
+
+// Type guard to check if filter key is valid for array operations
+const isArrayFilterKey = (filterKey: string): filterKey is ArrayFilterKey =>
+  isEnumMember(filterKey, ModelCatalogStringFilterKey) &&
+  filterKey !== ModelCatalogStringFilterKey.USE_CASE;
 
-export const useCatalogStringFilterState = (
-  filterKey: ModelCatalogStringFilterKey,
+export const useCatalogStringFilterState = <K extends ArrayFilterKey>(
+  filterKey: K,
 ): {
-  isSelected: (value: ModelCatalogStringFilterValueType[ModelCatalogStringFilterKey]) => boolean;
-  setSelected: (
-    value: ModelCatalogStringFilterValueType[ModelCatalogStringFilterKey],
-    selected: boolean,
-  ) => void;
+  isSelected: (value: ArrayFilterValueType[K]) => boolean;
+  setSelected: (value: ArrayFilterValueType[K], selected: boolean) => void;
 } => {
-  type Value = ModelCatalogStringFilterValueType[ModelCatalogStringFilterKey];
+  type Value = ArrayFilterValueType[K];
   const { filterData, setFilterData } = React.useContext(ModelCatalogContext);
-  const selections: Value[] = filterData[filterKey];
-  const isSelected = React.useCallback((value: Value) => selections.includes(value), [selections]);
+  const selections = filterData[filterKey];
+
+  const isSelected = React.useCallback(
+    (value: Value) => {
+      if (!isArrayOfValues<Value>(selections)) {
+        return false;
+      }
+      return selections.includes(value);
+    },
+    [selections],
+  );
+
   const setSelected = (value: Value, selected: boolean) => {
+    if (!isArrayOfValues<Value>(selections)) {
+      return;
+    }
+
     const nextState: Value[] = selected
       ? [...selections, value]
       : selections.filter((item) => item !== value);
-    if (isStringFilterValid(filterKey, nextState)) {
-      setFilterData(filterKey, nextState);
+
+    if (isArrayFilterKey(filterKey)) {
+      // Type assertion is safe here because we've verified the key is an array filter
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+      setFilterData(filterKey, nextState as ModelCatalogFilterStates[K]);
     }
   };
```

Comment:
Yeah, we'll want to revisit this - with type guards we shouldn't need an assertion like this. with the above reverted we probably can just revert this.